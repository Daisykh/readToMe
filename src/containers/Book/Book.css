img {
  width: auto;
  height: 50vh;
}

.Book {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.button-display {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

button {
  width: 75px;
  height: 30px;
}

.recording
 {
  color: white;
  box-shadow: none;
  border-style: none;
  border-radius: 5px;
  margin:0px 15px;
  border-radius:0.2em;
  background: #D60000; /* Old browsers */
  background: -moz-linear-gradient(left, #FF9100 1%, #D60000 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, #FF9100 1%, #D60000 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, #FF9100 1%,#D60000 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */
}

#disabled-record {
  opacity: 0.2; 
  cursor: not-allowed;
  color: white;
  box-shadow: none;
  border-style: none;
  border-radius: 5px;
  margin:0px 15px;
  border-radius:0.2em;
  background: #66e3e1; /* Old browsers */
  background: -moz-linear-gradient(left, #66e3e1 1%, #56c5e0 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, #66e3e1 1%,#56c5e0 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, #66e3e1 1%,#56c5e0 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */ 
}

#disabled-submit {
  opacity: 0.2; 
  cursor: not-allowed;
  color: white;
  box-shadow: none;
  border-style: none;
  border-radius: 5px;
  margin:0px 15px;
  border-radius:0.2em;
  background: #66e3e1; /* Old browsers */
  background: -moz-linear-gradient(left, #66e3e1 1%, #56c5e0 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, #66e3e1 1%,#56c5e0 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, #66e3e1 1%,#56c5e0 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */ 
}

.enabled-record {
  color: white;
  box-shadow: none;
  border-style: none;
  border-radius: 5px;
  margin:0px 15px;
  border-radius:0.2em;
  background: #66e3e1; /* Old browsers */
  background: -moz-linear-gradient(left, #66e3e1 1%, #56c5e0 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, #66e3e1 1%,#56c5e0 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, #66e3e1 1%,#56c5e0 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */
}

.enabled-record:hover {
  background: #D60000; /* Old browsers */
  background: -moz-linear-gradient(left, #FF9100 1%, #D60000 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, #FF9100 1%, #D60000 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, #FF9100 1%,#D60000 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */
}

.enabled-submit {
  color: white;
  box-shadow: none;
  border-style: none;
  border-radius: 5px;
  margin:0px 15px;
  border-radius:0.2em;
  background: #66e3e1; /* Old browsers */
  background: -moz-linear-gradient(left, #66e3e1 1%, #56c5e0 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, #66e3e1 1%,#56c5e0 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, #66e3e1 1%,#56c5e0 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */
}

.enabled-submit:hover {
  background: #D60000; /* Old browsers */
  background: -moz-linear-gradient(left, teal 1%, green 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, teal 1%, green 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, teal 1%, green 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */
}

.successful-submit {
  color: white;
  box-shadow: none;
  border-style: none;
  border-radius: 5px;
  margin:0px 15px;
  border-radius:0.2em;
  transition: 300ms all;
  background: green; /* Old browsers */
  background: -moz-linear-gradient(left, #66e3e1 1%, green 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left, #66e3e1 1%, green 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right, #66e3e1 1%, green 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#66e3e1', endColorstr='#56c5e0',GradientType=1 ); /* IE6-9 */
}


.enabled-record:focus,
.enabled-submit:focus,
.disabled-record:focus,
.disabled-submit:focus,
.recording:focus,
.successful-submit:focus
{
  outline: 0;
}


