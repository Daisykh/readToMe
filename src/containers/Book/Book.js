import React, { Component } from 'react';
import './Book.css'
const getUserMedia = require('get-user-media-promise');
const MicrophoneStream = require('microphone-stream');
// const socket = io('http://localhost:5000/');
const io = require('socket.io-client');

class Book extends Component {
  constructor(props) {
    super(props)

    this.state = {
      recording: false,
      recorded: false, 
      submitted: false,
      feedbackGiven: false,
    }

    this.micStream 
  }

  createMic = () => {
     this.micStream = new MicrophoneStream();
  }

  recordAudio = async () => {
    console.log('recording')
    const testSocket = io.connect('http://localhost:5000');
    testSocket.on('connect', (data) => {
      testSocket.emit('join', 'Hello World from client');
    })

    this.createMic();
    console.log('record selected');
    const micStream = this.micStream
    getUserMedia({ video: false, audio: true })
      .then(function(stream) {
       micStream.setStream(stream);
      }).catch(function(error) {
        console.log(error);
      })

    this.micStream.on('data', function(chunk) {
      const raw = MicrophoneStream.toRaw(chunk);
      console.log(raw)
    })
  }

  submitAudio = async () => {
   await console.log('submit selected');
   await this.toggleSubmit();
   await this.userFeedback();
  }

  toggleSubmit = () => {
    this.setState({
      submitted: true
    })
  }

  userFeedback = async () => {
    alert('Good job! You read the passage correctly');
    this.resetRecord();
    this.setState({
      feedbackGiven: true
    })
  }

  resetRecord = () => {
    this.setState({
      recording: false,
      recorded: false, 
    })
  }

  toggleRecord = () => {
    const recording = !this.state.recording;
    console.log('toggled')
    this.setState({ 
      recording: recording,
      submitted: false,
      feedbackGiven: false
    }, this.checkAudio);
  }

  checkAudio = () => {
    if (this.state.recording) {
      this.recordAudio();
    }

    if (!this.state.recording) {
      this.stopAudio();
    }
  }

  stopAudio = () => {
    console.log('stop selected')
    this.micStream.stop()
    this.setState({
      recorded: true
    })
  }

  render() {
    const url = this.props.img
    const activeRecord = this.state.recording ? "recording" : "enabled-record" ;
    const disableRecord = this.state.recorded && !this.state.recording ? "disabled-record": "";
    const disabledSubmit = this.state.recorded && !this.state.recording ? "" : "disabled-submit";
    const successfulSubmit = this.state.submitted ? "successful-submit" : "enabled-submit"

    return (
      <div className="Book">

        <img src={url} alt="book display"/>
        <p>{this.props.text}</p>
        <div className="button-display">
          <button onClick={ this.toggleRecord } id={disableRecord} className={activeRecord} >Read To Me!</button>
          <button onClick={ this.submitAudio } id={disabledSubmit} className={successfulSubmit} >Check</button>
        </div>
      </div>
    );
  }
}

export default Book;