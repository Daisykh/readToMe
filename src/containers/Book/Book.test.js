import React from 'react';
import { shallow } from 'enzyme';
import Book from './Book';

describe('Book', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(<Book />);

    expect(renderedComponent).toMatchSnapshot();
  })
})
