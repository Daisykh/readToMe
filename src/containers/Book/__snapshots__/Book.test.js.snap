// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should match snapshot 1`] = `undefined`;
